src/snowflake/cli/plugins/object/stage/manager.py

```diff
@@ -86,6 +85,32 @@ def get(
             f"get {self.quote_stage_name(stage_path)} {self._to_uri(dest_directory)} parallel={parallel}"
         )
 
+    def get_recursive(
+        self, stage_path: str, dest_path: Path, parallel: int = 4
+    ) -> List[SnowflakeCursor]:
+        list_files_result = self.list_files(stage_path).fetchall()
+        files_on_stage = [f["name"] for f in list_files_result]
+
+        if stage_path.startswith("snow://"):
+            stage_path = stage_path[7:]
+
+        stage_parts_length = len(Path(stage_path).parts)
+        results = []
+        for file in files_on_stage:
+            dest_directory = dest_path / "/".join(
+                Path(file).parts[stage_parts_length:-1]
+            )
+            self._assure_is_existing_directory(Path(dest_directory))
```

Comment:
Does it mean the directory has to exists?

Reply:
Yes

Reply:
Shouldn't get make sure it creates the directories? If I want to use this piece of code in other place, does it mean I have also to create all directories?

Reply:
`_assure_is_existing_directory` is already existing function which checks if directory exists and creates it if not. Normal `get` uses the same function.